src/showingpreviously/cinemas/jackroe.py

```diff
@@ -0,0 +1,163 @@
+import re
+
+from datetime import datetime, timedelta
+from bs4 import BeautifulSoup
+
+import showingpreviously.requests as requests
+from showingpreviously.model import ChainArchiver, CinemaArchiverException, Chain, Cinema, Screen, Film, Showing
+from showingpreviously.consts import STANDARD_DAYS_AHEAD, UK_TIMEZONE, UNKNOWN_FILM_YEAR
+
+
+SHOWING_LIST_URL = 'https://www.jack-roe.co.uk/websales/sales/{cinema_code}/start?comingsoon=1'
+SHOWING_URL = 'https://www.jack-roe.co.uk/websales/sales/{cinema_code}/actual_book?perfcode={showing_code}'
+SHOWING_CODE_PATTERN = re.compile(r'book\?perfcode=(?P<showing_code>\d+)$')
+SHOWING_INFO_PATTERN = re.compile(r'at(?P<time>\s+\d+:\d+)\s+on\s+(?P<date>.+?)\s+showing in\s+(?P<screen_name>.+?)\s*<')
+FINISHED_TEXT = 'We are afraid that sales for the following performance have now finished'
+
+SCOTTCINEMAS_BASE_URL = 'https://www.scottcinemas.co.uk'
+SCOTTCINEMAS_CHAIN = Chain('Scott Cinemas')
+SCOTTCINEMAS_LOCATION_CODE_PATTERN = re.compile(r'/websales/sales/(?P<location_code>.+?)/actual_book')
+
+WTWCINEMAS_BASE_URL = 'https://wtwcinemas.co.uk'
+WTWCINEMAS_CHAIN = Chain('WTW Cinemas')
+WTWCINEMAS_LOCATION_CODE_PATTERN = re.compile(r'/websales/sales/(?P<location_code>.+?)/book')
+
+
+def get_response(url: str) -> requests.Response:
+    r = requests.get(url)
+    if r.status_code != 200:
+        raise CinemaArchiverException(f'Got status code {r.status_code} when fetching URL {url}')
+    return r
+
+
+def get_film_attributes(film_title: str) -> dict[str, any]:
+    attributes = {}
+    if 'SUBTITLED SCREENING' in film_title:
+        attributes['subtitled'] = True
+        film_title = film_title.replace('SUBTITLED SCREENING', '')
+    if '(DUBBED)' in film_title:
+        attributes['language'] = 'English'
+        film_title = film_title.replace('(DUBBED)', '')
+    if film_title.endswith('2D'):
+        attributes['format'] = [' 2D']
+        film_title = film_title[:-3]
+    if film_title.endswith('3D'):
+        attributes['format'] = [' 3D']
```

Comment:
Why are we adding a space before the 2D / 3D format attributes? It should be "2D" and "3D", not " 2D" and " 3D", right?